safehttp/form.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"fmt"
+	"mime/multipart"
+	"strconv"
+)
+
+// Form contains parsed data either from URL's query or form parameters, part of
+// the body of POST, PATCH or PUT requests that are not  multipart requests. The
+// form values will only be available after parsing the form, and only through
+// the getter functions.
+type Form struct {
+	values map[string][]string
+	err    error
+}
+
+// Int64 checks whether key paramKey maps to any query or form parameter
+// values. In case it does, it will try to convert the first value to a 64-bit
```

Comment:
Why "query or form"? This is because you reuse the `Form` type for query params, right? That seems quite confusing. :|

Can we please focus on just `Form` and `MultiForm` in this PR?

Here and the other places.

Reply:
Yes.